redis_elasticache_client.go

```diff
@@ -48,13 +48,13 @@ func (r *redisElastiCacheDB) Initialize(_ context.Context, req dbplugin.Initiali
 		accessKey = r.config.Username
 	}
 	secretKey := r.config.SecretAccessKey
-	if secretKey == "" && r.config.SecretAccessKey != "" {
-		secretKey = r.config.SecretAccessKey
+	if secretKey == "" && r.config.Password != "" {
+		secretKey = r.config.Password
```

Comment:
@maxcoulombe just wanted to confirm this was the intention before I merge.
Thanks!

Reply:
Whoops, this is definitely a mistake.

This looks good to me. To be sure, in plain english:
* If the secretKey field is provided it has precedence
* If not, we try to look for the deprecated password field
* If both are unspecified we leave it blank so the util library fallbacks on the default provider chain e.g. try to load the creds from env vars